RevenueCatUI/CustomerCenter/ViewModels/VirtualCurrencyBalancesScreenViewModel.swift

```diff
@@ -0,0 +1,111 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  VirtualCurrenciesViewModel.swift
+//
+//  Created by Will Taylor on 4/21/25.
+
+#if os(iOS)
+
+import Foundation
+import RevenueCat
+
+/// A view model that manages the state and data for the virtual currencies screen.
+///
+/// This view model is responsible for loading and managing virtual currency balance data
+/// from RevenueCat's CustomerInfo.
+@available(iOS 15.0, *)
+@available(macOS, unavailable)
+@available(tvOS, unavailable)
+@available(watchOS, unavailable)
+final class VirtualCurrencyBalancesScreenViewModel: ObservableObject {
+
+    /// The current state of the virtual currencies screen.
+    @Published var viewState: VirtualCurrencyBalancesScreen.ViewState
+
+    /// A flag indicating whether the view model is running in a SwiftUI preview.
+    ///
+    /// When true, data loading is skipped to prevent unnecessary network calls
+    /// during preview rendering.
+    private let isRunningInSwiftUIPreview: Bool
```

Comment:
I would rather just inject a mock for `CustomerCenterPurchasesType` than adding this one, or just injecting a function that can easily be mocked 🤓

Reply:
Yeah, having a `isRunningInSwiftUIPreview` param isn't ideal. I didn't go with the mock route because `MockCustomerCenterPurchases` is only available in the `RevenueCatUITestsDev` target, not in the `RevenueCatUI` target. I guess we could move it into the `RevenueCatUI` target and keep it internal - do you have strong opinions either way?

Reply:
@ajpallares Had a good idea to compute `isRunningInSwiftUIPreview ` at runtime using `ProcessInfo.processInfo.environment["XCODE_RUNNING_FOR_PREVIEWS"]`, which allows us to remove `isRunningInSwiftUIPreview` from the initializer 😄. I've implemented that in 98af2c0d883a58ea7b24f1da8b8df67b10fcd13b

Reply:
I like this :)